lib/KaApi.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+// Copyright 2022. Plesk International GmbH. All rights reserved.
+
+namespace WHMCS\Module\Server\Plesk360Monitoring;
+
+use GuzzleHttp\Client;
+
+class KaApi
+{
+    private const CONTENT_TYPE_JSON = 'application/json';
+
+    private $client;
+
+    public function __construct(array $params)
+    {
+        $port = (int)$params[Params::SERVER_PORT] ?? 0;
+        $baseUrl = $params[Params::SERVER_SCHEME] . '://' . $params[Params::SERVER_HOST];
+
+        if ($port > 0) {
+            $baseUrl .= ':' . $port;
+        }
+
+        $username = $params[Params::SERVER_USERNAME];
+        $password = $params[Params::SERVER_PASSWORD];
+        $auth = 'Basic ' . base64_encode($username . ':' . $password);
+
+        $this->client = new Client([
+            'base_uri' => $baseUrl,
+            'base_url' => $baseUrl,
+            'defaults' => [
+                'headers' => [
+                    'Accept' => self::CONTENT_TYPE_JSON,
+                    'Authorization' => $auth,
+                    'Content-Type' => self::CONTENT_TYPE_JSON,
+                ],
+            ],
+            'headers' => [
+                'Accept' => self::CONTENT_TYPE_JSON,
+                'Authorization' => $auth,
+                'Content-Type' => self::CONTENT_TYPE_JSON,
+            ],
+        ]);
+    }
+
+    public function getLicenses(): array
+    {
+        $response = $this->client->get('/jsonrest/business-partner/30/keys');
+        $json = $response->getBody()->getContents();
+
+        return json_decode($json, true);
+    }
+
+    public function createLicense(Plan $plan, int $servers, int $websites): array
+    {
+        $options = [
+            'json' => [
+                'items' => $this->buildItems($plan, $servers, $websites),
+            ],
+        ];
+
+        $response = $this->client->post('/jsonrest/business-partner/30/keys?return-key-state=yes', $options);
+        $json = $response->getBody()->getContents();
+
+        return json_decode($json, true);
```

Comment:
i'd rather map it to License object

Reply:
this is optional

Reply:
in case not an object, then you can specify structure in the doc block of resulting array

Reply:
true for all other requests